src/index.ts

```diff
@@ -109,11 +109,14 @@ class DebuggerHandler<H extends ConsoleHandler | NotebookHandler> {
     });
 
     debug.model.disposed.connect(async () => {
-      await debug.stop();
-      Object.keys(this.handlers).forEach(id => {
-        this.handlers[id].dispose();
-      });
-      this.handlers = {};
+      const idsOfHandler = Object.keys(this.handlers);
+      if (idsOfHandler.length > 0) {
+        await debug.stop();
+        idsOfHandler.forEach(id => {
+          this.handlers[id].dispose();
+        });
+        this.handlers = {};
+      }
```

Comment:
How about using an early return if there is no handler?

```typescript
if (handlerIds.length === 0) {
  return;
}
await debug.stop();
handlerIds.forEach(id => {
  this.handlers[id].dispose();
});
this.handlers = {};

```

Reply:
Resolved